### ISSUE TITLE
Server Crashes randomly

### ISSUE THREAD

**Issue by zestym3m3s** [GC-spigot/AdvancedEnchantments#1904]
## Details

**Describe the bug**
In the last 2 days, since updating to the newest version of the plugin I've encountered like 4 crashes, usually when people are using items that place/destroy blocks such as trench and lavawalker. I'm not sure if they're related at all to the issue but it seems like theres some correlation. I've tried turning on and off use experimental block breaking but it happens regardless so I'm pretty sure its unrelated to that. Based on the crash reports it seems like something to do with lootpopulation or whatever that means.

1st crash I didn't know that anything was super weird
2nd crash one of my players reported jumping around with whatever the jump boots enchant is while on lava with lavawalker.
3rd crash one of my players entered a nether portal and 2 seconds later the server crashed.
4th crash I was wearing the pair of boots that was mentioned in the second crash with the waterwalking ae enchant that turns water into obsidian and the second I stepped on the water I saw a small T-tetris-shape of obsidian and the server crashed.

**To Reproduce**
I have no clue, based on the conditions that happened during each crash I hope it can help point you to what an issue might be.

**Screenshots / Video**
No screenshots or video but I have logs.

## Server Information
- "/ae plinfo" link: My servers currently offline since we can restart it without the host turning back on the PC. My username is zestym3m3s and I purchased the plugin on the advanced plugins website. Another name I might have is zesT and I'm zesT on the discord. I'm running on the 8.0.0 pre 23.8 version.
- Server log:
Time: 4/25/21 9:24 PM
Description: Exception in server tick loop

java.lang.IllegalAccessError: tried to access class net.advancedplugins.ae.handlers.lootPopulation.PopulationHandler$1 from class net.advancedplugins.ae.handlers.lootPopulation.PopulationHandler
	at net.advancedplugins.ae.handlers.lootPopulation.PopulationHandler.getLoot(PopulationHandler.java:73)
	at net.advancedplugins.ae.handlers.lootPopulation.listeners.WorldLootPopulator.generateLoot(WorldLootPopulator.java:43)
	at net.advancedplugins.ae.handlers.lootPopulation.listeners.WorldLootPopulator.populate(WorldLootPopulator.java:31)
	at net.minecraft.server.v1_16_R3.Chunk.loadCallback(Chunk.java:571)
	at net.minecraft.server.v1_16_R3.PlayerChunk.lambda$13(PlayerChunk.java:413)
	at net.minecraft.server.v1_16_R3.PlayerChunkMap$CallbackExecutor.run(PlayerChunkMap.java:143)
	at net.minecraft.server.v1_16_R3.ChunkProviderServer$a.executeNext(ChunkProviderServer.java:600)
	at net.minecraft.server.v1_16_R3.ChunkProviderServer.runTasks(ChunkProviderServer.java:294)
	at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1060)
	at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1044)
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.awaitTasks(SourceFile:127)
	at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:1028)
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:960)
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$0(MinecraftServer.java:273)
	at java.lang.Thread.run(Unknown Source)

All of the crashes seem to have this error in common. I hope its an easy fix :)

Heres my plugin list:
   Running: CraftBukkit version 3042-Spigot-a25e8ed-3512408 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT) true
   Plugins: { wishingwell v0.2 com.rstoski.wishingwell.WishingWell [rstoski], FakeLogin v3.1.1* me.pramsing.FakeLogin.Main [], PlaceholderAPI v2.10.9 me.clip.placeholderapi.PlaceholderAPIPlugin [HelpChat], IP-Check v2.0.7* net.risenphoenix.ipcheck.IPCheck [], CommandToItem v2.5 com.leonardobishop.commandtoitem.CommandToItem [LMBishop], PlugMan v2.2.0 com.rylinaux.plugman.PlugMan [rylinaux, Entity303], AutoRestart v4.2.1* org.serversmc.autorestart.core.Main [ServersMC, DennisLysenko], MC-Cron v1.8* me.tade.mccron.Cron [The_TadeSK], LuckPerms v5.3.3 me.lucko.luckperms.bukkit.loader.BukkitLoaderPlugin [Luck], ClearLag v3.2.2 me.minebuilders.clearlag.Clearlag [bob7l], WorldBorder v1.9.10 (beta) com.wimbli.WorldBorder.WorldBorder [Brettflan], BuycraftX v12.0.6 net.buycraft.plugin.bukkit.BuycraftPlugin [Buycraft], Votifier v2.7.2 com.vexsoftware.votifier.NuVotifierBukkit [Ichbinjoe, blakeman8192, Kramer, tuxed], KiteBoard v3.2.3* net.kitesoftware.board.KiteBoardPlugin [], ProtocolLib v4.5.1 com.comphenix.protocol.ProtocolLib [dmulloy2, comphenix], VoidSpawn v1.19.2 com.endercrest.voidspawn.VoidSpawn [tcvs], KixsAutoAnnouncerPremium v1.2.2 com.kixmc.KixsChatAnnouncer.Main.Main [kixmc], MVdWPlaceholderAPI v2.5.1* be.maximvdw.placeholderapi.PlaceholderAPI [Maximvdw], yardsale v1.0 com.rstoski.yardsale.Yardsale [rstoski], Vault v1.7.3-b131 net.milkbowl.vault.Vault [cereal, Sleaker, mung3r], PlaceholderSIGN v2.0.0 com.sagaciousteam.PlaceholderSIGN.Main [SagaciousTEAM], SimpleRename v13.8 io.github.galaipa.sr.SimpleRename [Galaipa, EnergizerBEAST1, ScarabCoder], Item2Chat v4.2.3 com.vk2gpz.item2chat.Item2Chat [vk2gpz], HolographicDisplays v2.4.6 com.gmail.filoghost.holographicdisplays.HolographicDisplays [filoghost], Factions v1.6.9.5-U0.5.21-b249 com.massivecraft.factions.FactionsPlugin [Olof Larsson, Brett Flannigan, drtshock, CmdrKittens], DynamicSigns v4.2.0* be.maximvdw.dsigns.DynamicSigns [Maximvdw], Drop-Party v3.3 us.divinerealms.neon.dropparty.DropParty [ampayne2], BeastTokens v3.9.9 me.mraxetv.beasttokens.BeastTokensPlugin [MrAxeTv], WorldEdit v7.2.2+fc3c499 com.sk89q.worldedit.bukkit.WorldEditPlugin [], CustomEnderChest v1.13.1 net.craftersland.customenderchest.EnderChest [CraftersLand], Essentials v2.18.2.0 com.earth2me.essentials.Essentials [Zenexer, ementalo, Aelux, Brettflan, KimKandor, snowleo, ceulemans, Xeology, KHobbits, md_5, Iaccidentally, drtshock, vemacs, SupaHam, md678685], VoteStreak v2.0 (for v1.13-1.14) de.aladram.votestreak.VoteStreak [Aladram], EssentialsChat v2.18.2.0 com.earth2me.essentials.chat.EssentialsChat [Zenexer, ementalo, Aelux, Brettflan, KimKandor, snowleo, ceulemans, Xeology, KHobbits, md_5, Okamosy, Iaccidentally], Parkour v6.4 io.github.a5h73y.parkour.Parkour [A5H73Y, steve4744], SuperbVote v0.5.5 io.minimum.minecraft.superbvote.SuperbVote [tuxed], TAB v2.8.10 me.neznamy.tab.platforms.bukkit.Main [NEZNAMY], PlotSquared v5.13.10-Premium com.plotsquared.bukkit.BukkitMain [Citymonstret, Empire92, MattBDev, dordsor21, NotMyFault, SirYwell], WorldGuard v7.0.4+f7ff984 com.sk89q.worldguard.bukkit.WorldGuardPlugin [], EssentialsProtect v2.18.2.0 com.earth2me.essentials.protect.EssentialsProtect [Zenexer, ementalo, Aelux, Brettflan, KimKandor, snowleo, ceulemans, Xeology, KHobbits], EssentialsGeoIP v2.18.2.0 com.earth2me.essentials.geoip.EssentialsGeoIP [Zenexer, ementalo, Aelux, Brettflan, KimKandor, snowleo, ceulemans, Xeology, kjiang], BanItem v3.0 fr.andross.banitem.BanItem [Andross], EssentialsAntiBuild v2.18.2.0 com.earth2me.essentials.antibuild.EssentialsAntiBuild [Zenexer, ementalo, Aelux, Brettflan, KimKandor, snowleo, ceulemans, Xeology, KHobbits, Iaccidentally], BetterRTP v3.0.5 me.SuperRonanCraft.BetterRTP.BetterRTP [SuperRonanCraft], EssentialsSpawn v2.18.2.0 com.earth2me.essentials.spawn.EssentialsSpawn [Zenexer, ementalo, Aelux, Brettflan, KimKandor, snowleo, ceulemans, Xeology, KHobbits], PvPManager v3.7.16 me.NoChance.PvPManager.PvPManager [NoChanceSD], WorldGuardExtraFlags v4.1.5-SNAPSHOT net.goldtreeservers.worldguardextraflags.WorldGuardExtraFlagsPlugin [isokissa3], BountyHunters v2.3.13 net.Indyuce.bountyhunters.BountyHunters [Indyuce], Multiverse-Core v4.2.2-b812 com.onarandombox.MultiverseCore.MultiverseCore [dumptruckman, Rigby, fernferret, lithium3141, main--], Citizens v2.0.27-SNAPSHOT (build 2247) net.citizensnpcs.Citizens [fullwall], mcMMO v2.1.173 com.gmail.nossr50.mcMMO [nossr50, GJ, NuclearW, bm01, Glitchfinder, TfT_02, t00thpick1, Riking, electronicboy, kashike], Multiverse-Portals v4.2.1-b834 com.onarandombox.MultiversePortals.MultiversePortals [Rigby, fernferret], dtlTradersPlus v6.3.3 com.degitise.minevid.dtlTraders.Main [Minevid, TheWGBbroz, samyratchet], SpecializedCrates v2.12.7 me.ztowne13.customcrates.SpecializedCrates [Ztowne13], Multiverse-Inventories v4.2.1-b490 com.onarandombox.multiverseinventories.MultiverseInventories [dumptruckman], Multiverse-SignPortals v4.2.0-b762 com.onarandombox.MultiverseSignPortals.MultiverseSignPortals [Rigby, fernferret], PlayerVaults v4.1.16 com.drtshock.playervaults.PlayerVaults [drtshock, kashike, CmdrKittens], PlayerHeads v5.2.12-SNAPSHOT org.shininet.bukkit.playerheads.PlayerHeads [meiskam, zand, crashdemons], Denizen v1.1.9-SNAPSHOT (build 1726-REL) com.denizenscript.denizen.Denizen [The DenizenScript Team], Jobs v4.17.0 com.gamingmesh.jobs.Jobs [phrstbrn, Zrips, montlikadani], ServerlistMOTD vX-2020-09-05* cloud.bolte.serverlistmotd.Main [Strumswell], CommandSigns v2.2.0 be.nokorbis.spigot.commandsigns.CommandSignsPlugin [Nokorbis], DiscordSRV v1.21.3 github.scarsz.discordsrv.DiscordSRV [Scarsz, Androkai, Vankka], Boss v3.9.7 org.mineacademy.boss.BossPlugin [kangarko], AuctionHouse v2.1.12 com.spawnchunk.auctionhouse.AuctionHouse [klugemonkey], Quests v4.0.2-b237 me.blackvein.quests.Quests [PikaMug], OreAnnouncer v2.7.2 com.alessiodp.oreannouncer.bukkit.bootstrap.BukkitOreAnnouncerBootstrap [AlessioDP], Statz v1.6.2 me.staartvin.statz.Statz [Staartvin], Autorank v4.5.1 me.armar.plugins.autorank.Autorank [Staartvin, Armarr], AdvancedEnchantments v8.0.0-Pre23.8 net.advancedplugins.ae.Core [GC],}

**Comment by zestym3m3s** [comment]
https://paste.md-5.net/uwuteqedef is the link, server just came online thankfully.

**Comment by Insprill** [comment]
could you also send the full latest.log?

**Comment by zestym3m3s** [comment]
[2021-04-26-3.log.gz](https://github.com/GC-spigot/AdvancedEnchantments/files/6380839/2021-04-26-3.log.gz)

**Comment by zestym3m3s** [comment]
and all four crashes have the same error in common

**Comment by Insprill** [comment]
can you update AE and see if it still happens? the Boss plugin loading one of our classes shouldn't be happening so it makes me think maybe that has something to do with it, but the error doesn't line up with the code so i can't really tell.

**Comment by zestym3m3s** [comment]
That was happening on the updated AE as well. I'm pretty sure that the AE which was loaded in that log was the 23.8 version. I just turned it back to an older version since I can't really have another crash tonight. I think that something has changed since 22.2 and its causing the crashes.

**Comment by zestym3m3s** [comment]
Is there any chance it has to do with random loot spawning in world chests? I noticed that theres a config for loot configuration and if I had to guess I feel like it might be related to that. The BOSS plugin isn't set up to spawn bosses yet so I'm not sure why it'd be doing something with loot generation. I'm no dev so I won't assume anything but I would like to see this get fixed so I can get some dope items working.
